preprocessors/openstreetmap/osm_service.py

```diff
@@ -178,119 +178,128 @@ def process_streets_data(OSM_data, bbox_coordinates):
+                # The condition above is true when the
+                # "bounded" list has just a node element.
+
+                # Variable index gives the position of this node in
+                # the unbounded_nodes.
                 index = unbounded_nodes.index(bounded_nodes[0])
-                if index > 0:  # if true, this first node element has
-                    # has a preceding node in the "unbounded_nodes" list.
-                    flag = False
-                    # so, update the "bounded_nodes" list with
-                    # the estimated value for the preceding node.
-                    bounded_nodes = add_new_node(
-                        flag, index, bounded_nodes,
-                        unbounded_nodes, bbox_coordinates)
-
-                # variable index gives the position of the last node element of
-                # the "bounded_nodes" list in the "unbounded_nodes" list.
-                index = unbounded_nodes.index(bounded_nodes[i - 1])
-                if index < j - 1:  # If true, this last node element has
-                    # has a succeeding node in the "unbounded_nodes" list.
-                    flag = True
-                    # so, update the "bounded_nodes" list with
-                    # the estimated value for the succeeding node.
-                    bounded_nodes = add_new_node(
-                        flag, index, bounded_nodes,
-                        unbounded_nodes, bbox_coordinates)
+                if (index < number_of_nodes_in_unbounded_nodes - 1 and
+                        index > 0):
+                    # The above condition is true for
+                    # a node element that has both the succeeding
+                    # and preceding nodes.
+                    there_is_a_succeeding_node = True
+                    if there_is_a_succeeding_node:
+                        # Estimate a value for the succeeding node.
+                        bounded_nodes = add_new_node(
+                            there_is_a_succeeding_node, index, bounded_nodes,
+                            unbounded_nodes, bbox_coordinates)
+                    there_is_a_preceding_node = True
+                    if there_is_a_preceding_node:
+                        # Equally estimate a value for the preceding node.
+                        there_is_a_succeeding_node = False
+                        bounded_nodes = add_new_node(
+                            there_is_a_succeeding_node, index, bounded_nodes,
+                            unbounded_nodes, bbox_coordinates)
+                elif index < number_of_nodes_in_unbounded_nodes - 1:
+                    # This above condition holds if
+                    # a node element has just the succeeding node.
+                    there_is_a_succeeding_node = True
+                    if there_is_a_succeeding_node:
+                        # Estimate a value for the succeeding node.
+                        bounded_nodes = add_new_node(
```

Comment:
Is there a way to move these repeated variable checks outside of the if-else structure here? For example, the same check and function exists above from lines 203 to 207. Splitting this out would improve maintainability.

Reply:
The flag variable is now taken outside the conditional if-else structure.

Reply:
The testing was done both on my local machine and on the unicorn. The test shows streets being represented as they should be, without ending prematurely prior to reaching the boundary, or extending beyond the boundary lines.  Also, the bad request (400) issue raised is equally addressed with this PR. So, now if a request is valid but does not meet the condition where OSM preprocessor should run, it will now return 204 and not 400.